File: apps/static-site/src/content/docs/1_guides/schema.ts
// schema.ts
import { z } from "zod";

import { createConsumerSchema } from "@rejot-dev/contract/consumer-schema";
import { createPublicSchema } from "@rejot-dev/contract/public-schema";

const myPublicSchema = createPublicSchema("my-public-schema", {
  source: { dataStoreSlug: "my-source-datastore" },
  outputSchema: z.object({
    id: z.string(),
    name: z.string(),
  }),
  config: {
    publicSchemaType: "postgres",
    transformations: [], // See next step
  },
  version: {
    major: 1,
    minor: 0,
  },
});

const myConsumerSchema = createConsumerSchema("my-consumer-schema", {
  source: {
    manifestSlug: "my-manifest",
    publicSchema: {
      name: "my-public-schema",
      majorVersion: 1,
    },
  },
  config: {
    consumerSchemaType: "postgres",
    destinationDataStoreSlug: "my-destination-datastore",
    sql: "", // See next step
  },
});

export default {
  myPublicSchema,
  myConsumerSchema,
};

File: apps/static-site/src/content/docs/1_guides/schema.py
# schema.py
from pydantic import BaseModel
from rejot_contract.public_schema import (
    create_public_schema, PublicSchemaConfig, Source, Version
)
from rejot_contract.consumer_schema import (
    create_consumer_schema, ConsumerSchemaConfig, SourceManifest, PublicSchema
)

class Account(BaseModel):
    id: int
    name: str


public = create_public_schema(
    public_schema_name="my-public-schema",
    source=Source(dataStoreSlug="my-source-datastore"),
    output_schema=Account,
    version=Version(major=1, minor=0),
    config=PublicSchemaConfig(
        publicSchemaType="postgres",
        transformations=[], # See next step
    ),
)

# Define a consumer schema
consumer = create_consumer_schema(
    "my-consumer-schema",
    source=SourceManifest(
        manifestSlug="my-manifest",
        publicSchema=PublicSchema(
            name="my-public-schema",
            majorVersion=1,
        ),
    ),
    config=ConsumerSchemaConfig(
        consumerSchemaType="postgres",
        destinationDataStoreSlug="my-destination-datastore",
        sql="", # See next step
    ),
)
